subprojects/base-services/src/main/java/org/gradle/internal/jvm/JdkTools.java

```diff
@@ -54,7 +54,9 @@ public static JdkTools current() {
         if (!javaVersion.isJava9Compatible()) {
             File toolsJar = javaInfo.getToolsJar();
             if (toolsJar == null) {
-                throw new IllegalStateException("Could not find tools.jar");
+                String javaHome = javaInfo.getJavaHome().getAbsolutePath();
```

Comment:
Just inline this call. We don't need the javaHome variable on any line other than the next one.